app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -122,372 +116,187 @@ public void run() {
-                                setmsg.setCurrent_cid(mAimsicdService.getCellTracker().getMonitorCell().getCID());
-                                setmsg.setCurrent_nettype(Device.getNetworkTypeName(mAimsicdService.getCell().getNetType()));
-                                int isRoaming = 0;
-                                if("true".equals(mAimsicdService.getCellTracker().getDevice().isRoaming())) {
-                                    isRoaming = 1;
-                                }
-                                setmsg.setCurrent_roam_status(isRoaming);
-                                //TODO is this the right place to get upto date geo location?
-                                setmsg.setCurrent_gps_lat(mAimsicdService.lastKnownLocation().getLatitudeInDegrees());
-                                setmsg.setCurrent_gps_lon(mAimsicdService.lastKnownLocation().getLongitudeInDegrees());
-
-                                //only alert if timestamp is not in the data base
-                                if(!dbacess.isTimeStampInDB(logcat_timestamp)) {
-                                    dbacess.storeCapturedSms(setmsg);
-
-                                    /*dbacess.insertEventLog(
-                                    MiscUtils.getCurrentTimeStamp(),
-                                            mAimsicdService.getCellTracker().getMonitorCell().getLAC(),
-                                            mAimsicdService.getCellTracker().getMonitorCell().getCID(),
-                                            mAimsicdService.getCellTracker().getMonitorCell().getPSC(),
-                                            String.valueOf(mAimsicdService.lastKnownLocation().getLatitudeInDegrees()),
-                                            String.valueOf(mAimsicdService.lastKnownLocation().getLatitudeInDegrees()),
-                                            (int)mAimsicdService.getCell().getAccuracy(),
-                                            3,
-                                            "Detected WAP PUSH sms"
-                                    );*/
-                                    dbacess.toEventLog(6, "Detected WAP PUSH (2) SMS");
-                                    MiscUtils.startPopUpInfo(tContext, 8);
-                                } else {
-                                    Log.d(TAG, "Detected SMS already logged");
-                                }
+    }
 
-                            }// end of if contains("SMS originating address:")
-                        }
-                        break;
-                    }
-                }
-            }
+    private void parseMwiSms(String[] bufflines,String logcat_timestamp){
+
+        CapturedSmsData setmsg = new CapturedSmsData();
+        String smstext = findSmsData(bufflines,null);
+        String num = findSmsNumber(bufflines,null);
+
+        if(smstext == null){smstext = "null";}
+        if(num == null){num = "null";}
```

Comment:
that's perfectly what I should do, but I expect you to learn something and not make the same mistake again and fix current mistakes.